roles/test_observabilityStrategy/tasks/main.yml

```diff
@@ -7,35 +7,42 @@
 - name: "Add observabilityStrategy: none to the spec section"
   ansible.builtin.shell:
     cmd: |
-      oc patch stf/default --type merge -p '{"spec":{"observabilityStrategy": "none"}}' && sleep 120
+      oc patch stf/default --type merge -p '{"spec":{"observabilityStrategy": "none"}}'
   changed_when: false
 
 
 - name: "Delete Grafana object"
   ansible.builtin.shell:
     cmd: |
-      oc delete grafana default && sleep 60
+      oc delete grafana default && while oc get po -l app=grafana | grep grafana; do sleep 1; done;
   changed_when: false
 
 
 - name: "Delete Prometheus object"
   ansible.builtin.shell:
     cmd: |
-      oc delete prometheus default && sleep 60
+      oc delete prometheus default && while oc get po -l app=prometheus | grep prometheus; do sleep 1; done;
   changed_when: false
 
 
 - name: "Delete Alertmanager object"
   ansible.builtin.shell:
     cmd: |
-      oc delete alertmanager default && sleep 60
+      oc delete alertmanager default && while oc get po -l app=alertmanager | grep alertmanager; do sleep 1; done;
   changed_when: false
 
 
 - name: "Delete ElasticSearch  object"
   ansible.builtin.shell:
     cmd: |
-      oc delete elasticsearches.elasticsearch.k8s.elastic.co elasticsearch && sleep 60
+      oc delete elasticsearches.elasticsearch.k8s.elastic.co elasticsearch &&  while oc get po | grep elasticsearch; do sleep 1; done;
+  changed_when: false
+
+
+- name: "Wait 2 minutes before checking that pods didn't return"
+  ansible.builtin.shell:
```

Comment:
There is an ansible module for this if you prefer https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pause_module.html

Reply:
Nice !